azavea/cac-tripplanner
# Add categories, activities, etc to place cards, pages. Add events UI.

## Overview

This PR integrates categories, activities, AWE membership, and wheelchair accessibility into place cards and place detail pages.

It also integrates events into place detail cards and introduces a new partial for event detail pages.

This PR is HTML and SASS only, and will require application-level changes to wire up the new features (see Notes below).


### Demo

#### Place card

##### Place, without origin

![image](https://user-images.githubusercontent.com/128699/34313319-3eb969c2-e738-11e7-8014-3bf25e2f72d0.png)

##### Place, with origin

![image](https://user-images.githubusercontent.com/128699/34313332-60ae7086-e738-11e7-92bb-ad7e9ef0a59e.png)

##### Event, without origin

![image](https://user-images.githubusercontent.com/128699/34313365-92e3caba-e738-11e7-9d86-e650fcbc9280.png)

##### Event, with origin

![image](https://user-images.githubusercontent.com/128699/34313358-8a194c02-e738-11e7-893c-5b40304e7707.png)


#### Place detail page

##### Desktop

![image](https://user-images.githubusercontent.com/128699/34313380-ade2bb0a-e738-11e7-9996-5ac07f2b1af5.png)

##### Mobile

![image](https://user-images.githubusercontent.com/128699/34313400-c36779d4-e738-11e7-914e-a46fb57f8842.png)



### Notes

#### Event cards
I considered generalizing the `.place-card` class to something like `.preview-card`, since they'll be used for both places and events. But`.preview-card` is already in use elsewhere in the app and I ultimately decided to leave it as is. 

When a `.place-card` is used for an event, the `.event-card` class must be added to the `.place-card` element, and the date and time elements should be populated.

#### New images
`awe-logo.png` and `awe-logo@2x.png` need to be wired up to the `src` and `srcset` attributes of `.place-awe-link .image` in `destination-detail.html`.

`awe-icon.png` and `awe-icon@2x.png` need to be wired up to the `src` and `srcset` attributes of `.place-card-badges .image` in `home.html` and `cac-home-templates.js`.

#### Activities
I've added icons for bike trails (aka cycling) and wheelchair accessibility (not an activity per se, but we're including it with the activity icons in the UI) to the fontello bundle.

I can add additional icons as additional activities are determined.

#### Event detail page
I created `partials/event-detail.html`, based on `partials/destination-detail.html`. I reckon we can use the `place-detail.html` template for both place and event detail pages, and just include the appropriate partial.

I changed all the `destination.*` variables to `event.*` and added some new ones (eg, `event.location`, `event.facebook_url`). Consider these pseudocode to get the point across. 

Note the event date and time are broken down into components, for styling.

Once this new partial is wired up to render events, I can do a follow-up PR to style the page.


## Testing Instructions

 * Clear your cache.
 * Ensure cards on the home page look like the images above.
   * Add the `.event-card` class to a `.place-card` to force it into event mode.
 * Go to a place's detail page. Ensure it looks like the images above.
 * Note the event partial can't be tested in this PR.


Connects #934, #935
